src/main/java/org/sonar/plugins/checkstyle/metadata/CheckstyleMetadata.java

```diff
@@ -100,9 +100,10 @@ public void createRulesWithMetadata() {
                     final SonarRulePropertyLoader.AdditionalRuleProperties additionalDetails =
                             additionalRuleData.get(check.getKey());
                     final RulesDefinition.NewRule rule =
-                            repository.createRule(moduleDetails.getFullQualifiedName());
+                            repository.createRule(moduleDetails.getFullQualifiedName()
+                                    + "template");
```

Comment:
the only minor item it have for this line.
it make rule id looks like 
`com.puppycrawl.tools.checkstyle.checks.naming.AbbreviationAsWordInNameChecktemplate`
but probably better to do
`com.puppycrawl.tools.checkstyle.checks.naming.AbbreviationAsWordInNameCheckTemplate`

@muhlba91 , please share your thoughts on this.